src/components/select/Select.js

```diff
@@ -1076,37 +1076,94 @@ export default class SelectComponent extends Field {
   }
 
   normalizeSingleValue(value) {
-    const dataType = _.get(this.component, 'dataType', 'auto');
+    if (!value) {
+      return;
+    }
 
-    switch (dataType) {
-      case 'auto':
-        if (!isNaN(parseFloat(value)) && isFinite(value)) {
-          value = +value;
+    const dataType = this.component['dataType'] || 'auto';
+    const denormalizedValue = typeof value === 'string' ? value.toLowerCase() : value;
+    const normalize = {
```

Comment:
Get rid of unnecessary try-catch statements, reduce code duplicates (`return this;` in try and catch branches), add support for object if needed.